service/start.go

```diff
@@ -280,10 +280,47 @@ func (svc *service) launchLNBackend(ctx context.Context, encryptionKey string) e
 		LNDMacaroonHex, _ := svc.cfg.Get("LNDMacaroonHex", encryptionKey)
 		lnClient, err = lnd.NewLNDService(ctx, svc.eventPublisher, LNDAddress, LNDCertHex, LNDMacaroonHex)
 	case config.LDKBackendType:
-		Mnemonic, _ := svc.cfg.Get("Mnemonic", encryptionKey)
-		LDKWorkdir := path.Join(svc.cfg.GetEnv().Workdir, "ldk")
+		mnemonic, _ := svc.cfg.Get("Mnemonic", encryptionKey)
+		ldkWorkdir := path.Join(svc.cfg.GetEnv().Workdir, "ldk")
+
+		nodeLastStartTime, _ := svc.cfg.Get("NodeLastStartTime", "")
+
+		// for brand new nodes, consider enabling VSS
+		if nodeLastStartTime == "" && svc.cfg.GetEnv().LDKVssUrl != "" {
+			albyUserIdentifier, err := svc.albyOAuthSvc.GetUserIdentifier()
+			if err != nil {
+				logger.Logger.WithError(err).Error("Failed to fetch alby user identifier")
+				return err
+			}
+			if albyUserIdentifier != "" {
+				me, err := svc.albyOAuthSvc.GetMe(ctx)
+				if err != nil {
+					logger.Logger.WithError(err).Error("Failed to fetch alby user")
+					return err
+				}
+				// only activate VSS for Alby paid subscribers
+				if me.Subscription.Buzz {
+					svc.cfg.SetUpdate("LdkVssEnabled", "true", "")
+				}
+			}
+		}
+
+		vssToken := ""
+		vssEnabled, _ := svc.cfg.Get("LdkVssEnabled", "")
+		if vssEnabled == "true" {
+			vssNodeIdentifier, err := ldk.GetVssNodeIdentifier(svc.keys)
+			if err != nil {
+				logger.Logger.WithError(err).Error("Failed to get VSS node identifier")
+				return err
+			}
+			vssToken, err = svc.albyOAuthSvc.GetVssAuthToken(ctx, vssNodeIdentifier)
```

Comment:
is this somehow possible to make it generic here that people could potentially self-host VSS?

Reply:
We decided for now to keep it as-is until we have a concrete usecase (running VSS for a single user may not be worthwhile for most users)